### ISSUE TITLE
ftrace record mcount build is noisy

### ISSUE THREAD

**Issue by shenki** [ClangBuiltLinux/linux#79]
When bulding 4.19-rc4 (plus clang fixes) for powerpc with FTRACE enabled, we see a lot of unknown flag warnings. This is because each object is re-built in order to record the mcount locations for dynamic ftrace, and that build process does not use KBUILD_CPPFLAGS.

`scripts/Makefile.build`:
```
sub_cmd_record_mcount = set -e ; perl $(srctree)/scripts/recordmcount.pl "$(ARCH)" \
        "$(if $(CONFIG_CPU_BIG_ENDIAN),big,little)" \
        "$(if $(CONFIG_64BIT),64,32)" \
        "$(OBJDUMP)" "$(OBJCOPY)" "$(CC)  $(KBUILD_CFLAGS)" \
        "$(LD) $(KBUILD_LDFLAGS)" "$(NM)" "$(RM)" "$(MV)" \
        "$(if $(part-of-module),1,0)" "$(@)";
```

We can fix this by adding $(KBUILD_CPPFLAGS) to the CC line.

**Comment by shenki** [comment]
Patch posted: https://lore.kernel.org/lkml/20180917073754.24048-1-joel@jms.id.au/

**Comment by nathanchance** [comment]
Merged into mainline as of 4.19-rc8: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=5a4630aadb9a9525474e9ac92965829f990cb5c4
